idunn/utils/opening_hours.py

```diff
@@ -37,16 +37,20 @@ def call(self, *args, **kwargs):
 
 class OpeningHours:
     def __init__(self, oh, tz, country_code):
-        if country_code is not None:
-            country_code = country_code.lower()
-
         self.raw = oh
         self.tz = tz
-        self.nmt_obj = {"address": {"country_code": country_code}}
+        if country_code:
+            self.nmt_obj = {"address": {"country_code": country_code.lower()}}
+        else:
+            self.nmt_obj = None
 
     def validate(self):
         """Check if an expression parses correctly"""
-        return engine.call("validate", self.raw, self.nmt_obj) is True
+        result = engine.call("validate", self.raw, self.nmt_obj)
+        if self.nmt_obj is not None and result is not True:
+            self.nmt_obj = None
+            return self.validate()
```

Comment:
I feels weird to hide any exception we could get when `nmt_obj` is not `None`, maybe it is possible to explicitly specify the kind of error we expect here?

Reply:
That's a good point.  

But openinghours.js throws raw error strings such as 
```
An error occurred during evaluation of the value "We-Su 17:00-21:00; PH off". Please submit a pull request: https://github.com/opening-hours/opening_hours.js. There are no holidays (PH) defined for country jp.
```
So I suppose we could catch the error and revalidate only if `"no holidays"` is contained in the string :worried: 
Any better idea ?